tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -174,6 +175,116 @@ DECL_CONVERT_OP(FakeQuant);
 
 #undef DECL_CONVERT_OP
 
+LogicalResult ConvertTFLGeluOp::matchAndRewrite(
+    Operation* op, PatternRewriter& rewriter) const {
+  auto tfl_gelu_op = cast<TFL::GeluOp>(op);
+  Location loc = op->getLoc();
+
+  Value input = tfl_gelu_op.input();
+  ShapedType input_type = input.getType().dyn_cast<ShapedType>();
+  ShapedType output_type =
+      tfl_gelu_op.getResult().getType().dyn_cast<ShapedType>();
+  // Not a ranked tensor output
+  if (!input_type || !output_type) return failure();
+
+  auto input_shape = input_type.getShape();
+  auto output_shape = output_type.getShape();
+
+  bool input_is_qtype =
+      input_type.getElementType().isa<mlir::quant::UniformQuantizedType>();
+  bool output_is_qtype =
+      output_type.getElementType().isa<mlir::quant::UniformQuantizedType>();
+
+  if (input_is_qtype != output_is_qtype) {
+    return rewriter.notifyMatchFailure(
+        op,
+        "ConvertTFLGeluOp: input/output tensor should be all quantized or all "
```

Comment:
Error convention is same as LLVM's as sentence fragment with no trailing period. Also can elide ConvertTFLGeluOp as the debugging output would include the message on the op and the pattern that failed to apply.

Reply:
Sorry, I didn't notice LLVM error convention before.
Done.